src/components/ReportActionItem/ReportPreview.tsx

```diff
@@ -193,63 +159,67 @@ function ReportPreview(props) {
+        [chatReport?.isOwnPolicyExpenseChat, policy?.isHarvestingEnabled],
     );
 
-    const getDisplayAmount = () => {
+    const getDisplayAmount = (): string => {
         if (hasPendingWaypoints) {
-            return props.translate('common.tbd');
+            return translate('common.tbd');
         }
         if (totalDisplaySpend) {
-            return CurrencyUtils.convertToDisplayString(totalDisplaySpend, props.iouReport.currency);
+            return CurrencyUtils.convertToDisplayString(totalDisplaySpend, iouReport?.currency);
         }
         if (isScanning) {
-            return props.translate('iou.receiptScanning');
+            return translate('iou.receiptScanning');
         }
         if (hasOnlyDistanceRequests) {
-            return props.translate('common.tbd');
+            return translate('common.tbd');
         }
 
         // If iouReport is not available, get amount from the action message (Ex: "Domain20821's Workspace owes $33.00" or "paid ₫60" or "paid -₫60 elsewhere")
         let displayAmount = '';
-        const actionMessage = lodashGet(props.action, ['message', 0, 'text'], '');
+        const actionMessage = action.message?.[0]?.text ?? '';
         const splits = actionMessage.split(' ');
-        for (let i = 0; i < splits.length; i++) {
-            if (/\d/.test(splits[i])) {
-                displayAmount = splits[i];
+
+        splits.forEach((split) => {
+            if (!/\d/.test(split)) {
+                return;
             }
-        }
+
+            displayAmount = split;
+        });
+
         return displayAmount;
     };
 
     const getPreviewMessage = () => {
         if (isScanning) {
-            return props.translate('common.receipt');
+            return translate('common.receipt');
         }
-        const payerOrApproverName = isPolicyExpenseChat ? ReportUtils.getPolicyName(props.chatReport) : ReportUtils.getDisplayNameForParticipant(managerID, true);
+        const payerOrApproverName = isPolicyExpenseChat ? ReportUtils.getPolicyName(chatReport) : ReportUtils.getDisplayNameForParticipant(managerID, true);
         if (isApproved) {
-            return props.translate('iou.managerApproved', {manager: payerOrApproverName});
+            return translate('iou.managerApproved', {manager: payerOrApproverName});
         }
-        const managerName = isPolicyExpenseChat ? ReportUtils.getPolicyName(props.chatReport) : ReportUtils.getDisplayNameForParticipant(managerID, true);
-        let paymentVerb = hasNonReimbursableTransactions ? 'iou.payerSpent' : 'iou.payerOwes';
-        if (iouSettled || props.iouReport.isWaitingOnBankAccount) {
+        const managerName = isPolicyExpenseChat ? ReportUtils.getPolicyName(chatReport) : ReportUtils.getDisplayNameForParticipant(managerID, true);
+        let paymentVerb: PaymentVerbTranslationPath = hasNonReimbursableTransactions ? 'iou.payerSpent' : 'iou.payerOwes';
```

Comment:
How about using TranslationPaths here instead?
```suggestion
        let paymentVerb: TranslationPaths = hasNonReimbursableTransactions ? 'iou.payerSpent' : 'iou.payerOwes';
```